src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -132859,6 +132877,47 @@ public static void Main()
             CompileAndVerify(executeComp, expectedOutput: "ran");
         }
 
+        [Fact, WorkItem(41437, "https://github.com/dotnet/roslyn/issues/41437")]
+        public void CannotAssignMaybeNullToTNotNull()
+        {
+            var source = @"
+class C
+{
+    TNotNull M<TNotNull>(TNotNull t) where TNotNull : notnull
+    {
+        if (t is null) System.Console.WriteLine();
+
+        return t; // 1
+    }
+}";
+            var comp = CreateNullableCompilation(source);
+            comp.VerifyDiagnostics(
+                // (8,16): warning CS8603: Possible null reference return.
+                //         return t; // 1
+                Diagnostic(ErrorCode.WRN_NullReferenceReturn, "t").WithLocation(8, 16)
+                );
+        }
+
+        [Fact]
+        public void TNotNullFieldMustBeAssigned()
```

Comment:
>TNotNullFieldMustBeAssigned [](start = 20, length = 27)

Was the behavior of this method affected? This case and others are covered in `UninitializedNonNullableFieldTests.GenericType_NotNullConstraint`.